nucypher/characters/lawful.py

```diff
@@ -962,6 +964,90 @@ def retrieve(self,
 
         return cleartexts
 
+    def _handle_treasure_map(self, treasure_map, relayer_verifying_key: PublicKey, label: bytes):
+        if treasure_map is not None:
+
+            if self.federated_only:
+                from nucypher.policy.maps import TreasureMap as _MapClass
+            else:
+                from nucypher.policy.maps import SignedTreasureMap as _MapClass
+
+            # TODO: This LBYL is ugly and fraught with danger.  NRN
+            if isinstance(treasure_map, bytes):
+                treasure_map = _MapClass.from_bytes(treasure_map)
+
+            if isinstance(treasure_map, str):
+                tmap_bytes = treasure_map.encode()
+                treasure_map = _MapClass.from_bytes(b64decode(tmap_bytes))
+
+            self._try_orient(treasure_map, relayer_verifying_key)
+            # self.treasure_maps[treasure_map.public_id()] = treasure_map # TODO: Can we?
+        else:
+            map_id = self.construct_map_id(relayer_verifying_key, label)
+            try:
+                treasure_map = self.treasure_maps[map_id]
+            except KeyError:
+                # If the treasure map is not known, join the policy as part of retrieval.
+                self.join_policy(label=label, alice_verifying_key=relayer_verifying_key)
```

Comment:
Should join_policy take a `relayer_verifying_key` as the parameter then instead of an `alice_verifying_key`?

Reply:
✔️ - In isolation I think this is more meaningful as "publisher" instead of "relayer".  what do you think?

Reply:
✔️ Renamed relayer -> publisher.